File: codeforces/1600/1654/C/second_try/C8.cpp
#include <bits/stdc++.h>

using namespace std;

using ll = long long;

int main() {
    int t; cin >> t;
    ll n;
    ll total;
    ll a_max, b_max;
    ll b_new1, b_new2;
    while(t--) {
        cin >> n;
        vector<ll> a (n);
        total = 0;
        for (int i = 0; i < n; i++) {
            cin >> a[i]; 
            total += a[i];
        }

        sort(a.begin(), a.end());

        vector<ll> b = {total};
        cout << b[0] << endl;

        while (a.size() > 0 && b.size() > 0) {
            a_max = a[a.size() - 1];
            b_max = b[b.size() - 1]; 

            if (a_max  > b_max ) {
                cout << "NO";
                break;
            } 

            if (a_max == b_max) {
                a.pop_back();
                b.pop_back();
            }

            if (a_max < b_max) {
                b_new1 = floor(b_max/2.0);
                b_new2 = ceil(b_max/2.0);
            }

        }
    }

    return 0;
}

File: codeforces/1600/1654/C/second_try/C8.py
from math import ceil, floor
from bisect import insort, bisect_left

T = int(input())

for t in range(T):
    n = int(input())
    a = [int(el) for el in input().split()]
    a = sorted(a)

    if len(a) < 2:
        print("YES")
        continue

    b = [sum(a)]

    safe = 100 
    iter = 0
    while (len(a) > 0 and len(b) > 0):
        if a[-1] > b[-1]:
            print("NO")
            break

        if b[-1] == a[-1]:
            b.pop(-1)
            a.pop(-1)
        else:
            el_over_two = b.pop(-1)/2
            b_new1, b_new2 = floor(el_over_two), ceil(el_over_two)
            idx1 = bisect_left(b, b_new1)
            b.insert(idx1, b_new1)
            insort(b, b_new2, lo=idx1)

    if (len(a) == 0):
        print("YES")

        iter += 1
        safe -= 1
        if safe <= 0:
            break




